### PR TITLE
Azure Maps Weather SDK

### PR DESCRIPTION
**Pull request by khmic5**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x ] **The pull request does not introduce [breaking changes]**
- [ x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ x] Title of the pull request is clear and informative.
- [ x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/maps/azure-maps-weather/README.md [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+
+```java com.azure.maps.weather.sync.builder.ad.instantiation
+// Authenticates using Azure AD building a default credential
+// This will look for AZURE_CLIENT_ID, AZURE_TENANT_ID, and AZURE_CLIENT_SECRET env variables
+DefaultAzureCredential tokenCredential = new DefaultAzureCredentialBuilder().build();
+
+// Creates a client
+WeatherClient client = new WeatherClientBuilder()
+    .credential(tokenCredential)
+    .weatherClientId(System.getenv("MAPS_CLIENT_ID"))
```
Comment (lmolkova) [inline sdk/maps/azure-maps-weather/README.md:60]: nit
```suggestion
    .weatherClientId(System.getenv("WEATHER_CLIENT_ID"))
```

**Review by lmolkova — APPROVED** [review]

File: sdk/maps/azure-maps-weather/README.md (hunk 1 above)
Comment (khmic5) [reply sdk/maps/azure-maps-weather/README.md:60]: In the setup for Azure Maps, the environment variable is maps_client_id across all modules including search, route render etc!

**Pull request merged**